lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -27,11 +30,10 @@ def deploy_succeeded?
                        "Consider switching to rollingUpdate.")
           @success_assumption_warning_shown = true
         end
+      else
+        success &= desired_replicas == status_data['currentReplicas'].to_i
```

Comment:
This property is not supported for STS with OnDelete:

```
kubectl rollout status sts/foo                                                                                                                                                                          
error: rollout status is only available for RollingUpdate strategy type
```

... therefore this condition can never be true in this case and krane will stall.

Reply:
@timothysmith0609 Great question!

I had to look into this a bit. The brief answer is certain combinations of operations can cause the `currentReplicas` field become `nil`. The simple test as-is does not hit these conditions, so it passes.

I've pushed a version of this test that hangs (in the same way I saw in real life) here: https://github.com/Shopify/krane/commit/5c4acd050aa06205ae38706bb81a8fcfefad2bb8

You can observe locally this is the state this sequence of events leaves the stateful set in:

```
  status:
    collisionCount: 0
    currentRevision: stateful-busybox-6b5489d7d6
    observedGeneration: 2
    readyReplicas: 2
    replicas: 2
    updateRevision: stateful-busybox-96676bd54
    updatedReplicas: 2
```

Notice the absence of `currentReplicas`.

I did not push this version of the test here because it's hanging due to [the](https://github.com/Shopify/krane/pull/876/files#r821258485) so the only way to runt his test is with `verify_result: false` which doesn't make for a good test.

If you prefer, I can drop these changes from this PR since IRL I will have to run without verify result anyway until the previously mentioned bug is fixed.

Reply:
>If you prefer, I can drop these changes from this PR since IRL I will have to run without verify result anyway for now.

May as well keep the changes for now, to avoid repeating the same work later.